js/cli/src/generate/index.ts

```diff
@@ -3,16 +3,141 @@ import { Sails } from 'sails-js';
 import { ServiceGenerator } from './service-gen.js';
 import { TypesGenerator } from './types-gen.js';
 import { Output } from './output.js';
+import { existsSync, mkdirSync, readFileSync, writeFileSync } from 'fs';
+import path from 'path';
+import { confirm } from '@inquirer/prompts';
+import * as config from '../config.json';
 
-export function generateLib(sails: Sails, className = 'Program'): string {
-  const out = new Output();
+export class ProjectBuilder {
+  private projectPath = ['.', 'src'];
+  private isProject: boolean = true;
+  private isAutomaricOverride: boolean = false;
 
-  const typesGen = new TypesGenerator(out, sails.program);
-  typesGen.generate();
+  constructor(private sails: Sails, private name: string = 'Program') {}
 
-  const serviceGen = new ServiceGenerator(out, sails.program, sails.scaleCodecTypes);
+  private async canCreateFile(filePath: string): Promise<boolean> {
+    if (this.isAutomaricOverride || !existsSync(filePath)) {
+      return true;
+    }
 
-  serviceGen.generate(className);
+    const answer = await confirm({
+      message: `File ${filePath} exists. Do you want to overwrite?`,
+      default: false,
+    });
 
-  return out.finalize();
+    return answer;
+  }
+
+  private generateLib(): string {
+    const out = new Output();
+
+    const serviceGen = new ServiceGenerator(out, this.sails.program, this.sails.scaleCodecTypes);
+    serviceGen.generate(this.name);
+
+    return out.finalize();
+  }
+
+  private generateTypes(): string {
+    const out = new Output();
+
+    const typesGen = new TypesGenerator(out, this.sails.program);
+    typesGen.generate();
+
+    return out.finalize();
+  }
+
+  setIdlPath(path: string) {
+    const idl = readFileSync(path, 'utf-8');
+    this.sails.parseIdl(idl);
+
+    return this;
+  }
+
+  setRootPath(path: string) {
+    this.projectPath[0] = path ? path : '.';
+
+    return this;
+  }
+
+  setIsProject(isProject: boolean) {
+    this.isProject = isProject;
+
+    return this;
+  }
+
+  setAutomaticOverride(isAutomaricOverride: boolean) {
+    this.isAutomaricOverride = isAutomaricOverride;
+
+    return this;
+  }
+
+  async build() {
+    const rootPath = this.projectPath[0];
+    const srcPath = path.join(...this.projectPath);
+
+    const libFilePath = this.isProject ? path.join(...this.projectPath) : this.projectPath[0];
```

Comment:
If I got it right this path is not the path to lib.ts file but the path to the directory where lib.ts file is generated. If so, let's rename it to make it clearer